One star task solution.

IgushkinV opened the pull request.

IgushkinV committed:
One star task solution.

IgushkinV committed:
One star task solution.
Class Sausage.java

tlt-java-courses commented on src/main/java/igushkin/homeworks/lesson10/task2/TaskUtilities.java:
```diff
@@ -0,0 +1,96 @@
+     */
+    public void oneStreamWrite(List<T> listOfObjects, Path path) {
+        Base64.Encoder encoder = Base64.getEncoder();
+        try (FileWriter writer = new FileWriter(path.toFile())) {
```
Давай использовать функционал Java 8/11 для считывания из файла

IgushkinV replied:
Использовал Files.lines().

tlt-java-courses commented on src/main/resources/sausage.txt:
```diff
@@ -0,0 +1,4 @@
+rO0ABXNyAClpZ3VzaGtpbi5ob21ld29ya3MubGVzc29uMTAudGFzazIuU2F1c2FnZbOyBNZ53T3mAgADSgAEY29zdEkABndlaWdodEwABHR5cGV0ABJMamF2YS9sYW5nL1N0cmluZzt4cAAAAAAAAAAAAAAAAHA=
```
А что это за файл?

IgushkinV replied:
Это файл, в который записываются закодированные объекты. И потом считываются из него.
Его не должно быть?

tlt-java-courses replied:
По заданию записываться во втором задании ничего не должно, был приложен файл к ДЗ

tlt-java-courses reviewed (changes requested):
Для_ чего пустой файл  `src/main/resources/null.txt`?

_Ответ: пустой файл сделал для того, чтобы показать, как будет работать считывание из него._

tlt-java-courses commented on src/test/java/igushkin/homeworks/lesson10/task2/TaskUtilitiesTest.java:
```diff
@@ -0,0 +1,55 @@
+    private static TaskUtilities<Sausage> taskUtilities;
+
+    @BeforeAll
+    public static void prepare() {
```
setUp метод обычно называют

IgushkinV replied:
Переименовал.

tlt-java-courses commented on src/test/java/igushkin/homeworks/lesson10/task2/TaskUtilitiesTest.java:
```diff
@@ -0,0 +1,55 @@
+ */
+class TaskUtilitiesTest {
+
+    private static List<Sausage> sausageList = new ArrayList<>();
```
static переменные в тестах не используют

IgushkinV replied:
Убрал.

tlt-java-courses commented on src/main/java/igushkin/homeworks/lesson10/task2/TaskUtilities.java:
```diff
@@ -0,0 +1,96 @@
+    public List<Optional<T>> oneStreamRead(Path path) throws IOException {
+        Base64.Decoder decoder = Base64.getDecoder();
+        log.debug("oneStreamRead() - Lines count: {}", Files.lines(path).count());
+        return Files.lines(path)
```
А что так сложно всё это?

IgushkinV replied:
Упростил, убрав промежуточный этап с байтами. Больше упрощений пока не вижу.

IgushkinV committed:
Исправление замечаний.
Реализация третьего задания на половину: уже собирает объектное представление файлов и папок по указанному пути, но еще не печатает.
